### ISSUE TITLE
Renamed MediatedTransfer to LockedTransfer

### ISSUE THREAD

**Issue by hackaugusto** [raiden-network/raiden#140]
The name `MediatedTransfer` implies that mediation is requried, but there are user-cases where a hashlocked transfer could be made among to nodes with a channel open. Rename the `MediatedTransfer` to  `LockedTransfer` to reflect the fact that it's locked with a hashlock but not necessarily mediated.

**Comment by LefterisJP** [comment]
What will happen to the already existing `LockedTransfer` then? It's a base class of `MediatedTransfer`. Is it just not needed?

**Comment by hackaugusto** [comment]
The existing `LockedTransfer` is just a python implementation detail, we use it to do type checking and some code reuse.
We could expand the classes that inherit from `LockedTransfer` and remove it or find a different name for either the `MediatedTransfer` or the `LockedTransfer`.

**Comment by konradkonrad** [comment]
Just a few thoughts here:
a `LockedTransfer` base class may still be useful, e.g. for creating a "`ConditionallyLockedTransfer`": where the unlocking is resolved through another process (like an oracle).

IIRC, the reason for this ticket was to open up hashlocked transfers for non-mediated cases. How about this?:

- make it possible to use a `HashLockedTransfer(LockedTransfer)` as a direct transfer.
- keep the mediation logic bound to instances of `MediatedTransfer(LockedTransfer)`.

**Comment by LefterisJP** [comment]
My biggest beef with the current setup is that a new user reading through the codebase has no way of determining that `LockedTransfer` is just an implementation detail. It's not even written in the [class's docstring](https://github.com/raiden-network/raiden/blob/master/raiden/messages.py#L385):

```python
class LockedTransfer(SignedMessage):
    """ A transfer which signs that the partner can claim `locked_amount` if
    she knows the secret to `hashlock`.

    The asset amount is implicitly represented in the `locksroot` and won't be
    reflected in the `transferred_amount` until the secret is revealed.

    This signs Carol, that she can claim locked_amount from Bob if she knows the secret to hashlock

    If the secret to hashlock becomes public, but Bob fails to sign Carol a netted balance,
    with an updated rootlock which reflects the deletion of the lock, then
    Carol can request settlement on chain by providing:
        any signed [nonce, asset, balance, recipient, locksroot, ...]
        along a merkle proof from locksroot to the not yet netted formerly locked amount
    """
```

It is even instantiated in [some parts](https://github.com/raiden-network/raiden/blob/master/raiden/tests/test_channel.py#L59) of the codebase which reinforces the idea that such a type of message exists. 

Granted, later down the code it is transformed into a `MediatedTransfer` but that's easy to miss.

If it's a code abstraction it should be treated as such.

**Comment by heikoheiko** [comment]
Yes, there will probably be multiple kinds of locked transfers in the future. So maybe rename `MediatedTransfer` to `LockedTransfer` and `LockedTransfer` to `LockedTransferBase`

**Comment by hackaugusto** [comment]
Related to #146

**Comment by LefterisJP** [comment]
@hackaugusto What's the status of this issue? Do you think it is stll relevant or is the current situation good enough?

**Comment by hackaugusto** [comment]
I still think that the mediated transfer name is a bit confusing, and the classes/messages were not totally renamed.

**Comment by LefterisJP** [comment]
I don't see the name going away soon and I think we got more important things to focus on. 

Plus it could be argued that we could sit down and think on naming and teminology for all components under the auspices of the [Terminology](https://github.com/raiden-network/raiden/issues/146) issue.

I see overlap between these two issues.
